FlowCrypt/Functionality/Api/Remote Pub Key Apis/PubLookup.swift

```diff
@@ -67,9 +67,11 @@ class PubLookup: PubLookupType {
         return try RecipientWithSortedPubKeys(recipient, keyDetails: attesterResult.keys)
     }
 
-    func fetchRemoteUpdateLocal(with recipient: Recipient) async throws -> RecipientWithSortedPubKeys {
-        let recipient = try await self.lookup(recipient: recipient)
-        try localContactsProvider.updateKeys(for: recipient)
-        return recipient
+    func fetchRemoteUpdateLocal(with recipient: Recipient) async throws {
+        let remoteRecipient = try await self.lookup(recipient: recipient)
+        if remoteRecipient.pubKeys.isEmpty {
+            throw ContactsError.keyMissing
```

Comment:
Added this throw when remote returns empty keys.
And it's handled here.
Let me know your thoughts @tomholub 

![image](https://user-images.githubusercontent.com/77344191/170115379-25629ea2-3ac4-4f37-aeda-6f863b67acd1.png)

Reply:
No public keys for a recipient is not an exceptional state, it's pretty normal. I wouldn't be throwing an exception for that. It's odd to see an exception handler for that, too. Normally, no public keys would be represented with an empty array.

In the code that I was commenting on, the result was optional and I was just wondering what would happen if it came back nil.

Reply:
Do you want me to revert this throw logic?

Reply:
Ok. Please revert that. Else looks ok, as long as this is true:

> If fetchRemoteUpdateLocal succeed, localContactsProvider.searchRecipient would not return nil

Reply:
ok